server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -609,57 +704,45 @@ private static SnapshotsInRepo sortSnapshots(
         @Nullable String snapshotName,
         @Nullable String repoName
     ) {
-        final Predicate<SnapshotInfo> isAfter;
         switch (sortBy) {
             case START_TIME:
-                isAfter = filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
-                break;
+                return filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
             case NAME:
                 if (snapshotName == null) {
-                    assert repoName == null : "no snapshot name given but saw repo name [" + repoName + "]";
-                    isAfter = order == SortOrder.ASC
-                        ? snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) <= 0
-                        : snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) >= 0;
+                    // already covered by preflight filtering
+                    return null;
```

Comment:
Maybe add the Nullable annotation to the returned value of the method.

Reply:
refactored this a little with Henning's suggestions now and added nullable wherever it was appropriate